dataplane/driver.go

```diff
@@ -297,6 +297,9 @@ func StartDataplaneDriver(configParams *config.Config,
 				InterfaceName:       configParams.WireguardInterfaceName,
 				MTU:                 configParams.WireguardMTU,
 				RouteSource:         configParams.RouteSource,
+				// currently, AKS *must* encrypt hostnetwork traffic, while it is optional on EKS
+				HostEncryptionEnabled: configParams.KubernetesProvider() == config.ProviderAKS ||
```

Comment:
I would consider extracting this logic in to a helper function to make it a bit more readable, but that's just my opinion.